cpp/harvest/nacjd.cc

```diff
@@ -178,6 +178,29 @@ bool DownloadID(std::ofstream &json_new_titles, const std::string &id, const boo
     return true;
 }
 
+void GetIDFromLOK(const MARC::Record &record, std::set<std::string> * const parsed_marc_ids) {
+    for (auto field(record.begin()); field != record.end(); ++field) {
```

Comment:
If you want to iterate over LOK and local data blocks, you should use the helper functions from MARC.h: `findStartOfAllLocalDataBlocks()` and `getLocalTagRange()`.

If you need some examples, you can have a look at e.g. `normalise_urls.cc` or `krimdok_check_local_holdings.cc`.